src/dscl.ts

```diff
@@ -0,0 +1,316 @@
+// https://stackoverflow.com/a/70994696/6183068
+const satisfies =
+  <T>() =>
+  <U extends T>(u: U) =>
+    u;
+
+const generateDsclPath: Fig.Generator = {
+  trigger: "/",
+  getQueryTerm: "/",
+  custom: async (tokens, executeShellCommand) => {
+    const lastToken = tokens[tokens.length - 1];
+    if (lastToken === "") {
+      return [];
+    }
+    // can't guarantee that dscl is the first token. If it's not found,
+    // this is -1, so searching for `datasource` will start from index 0
+    const dsclIndex = tokens.findIndex((token) => token === "dscl");
+    let datasource: string;
+    for (let i = dsclIndex + 1; i < tokens.length; i++) {
+      const token = tokens[i];
+      // skip if the current token is a flag
+      if (token.startsWith("-")) {
+        continue;
+      }
+      // skip if the previous token is a flag that takes an argument
+      const prev = tokens[i - 1];
+      if (
+        dsclOptions.find(
+          (option) => [option.name].flat().includes(prev) && !!option.args
+        )
+      ) {
```

Comment:
This is really clever!! We could extract this so we don't have to check it every loop, computed on load instead:
```ts

const dsclOptionsWithArgs = new Set<string>(
  dsclOptions
    .filter((option) => option.args !== undefined)
    .flatMap((option) => option.name),
);

const generateDsclPath: Fig.Generator = {
  // ...
```

Then we can just check the Set:

```suggestion
      if (dsclOptionsWithArgs.has(prev)) {
```

Reply:
Good suggestion! 😄